Book/php7/classes_objects/object_handlers.rst

```diff
@@ -0,0 +1,137 @@
+Object handlers
+===============
+
+Nearly all operations on objects in PHP go through object handlers and every magic method or magic interface is
+implemented with an object or class handler internally. Furthermore there are quite a few handlers which are not exposed
+to userland PHP. For example internal classes can have custom comparison and cast behavior.
+
+As the number of different object handlers is rather large we can only discuss examples (using the typed array
+implementation from the last section) for a few of them. For all the others only a short description is provided.
+
+An Overview
+-----------
+
+Here are all the object handlers with their signature and a small description.
+
+.. c:member::
+    zval *read_property(zend_object *object, zend_string *member, int type, void **cache_slot, zval *rv)
+    zval *write_property(zend_object *object, zend_string *member, zval *value, void **cache_slot)
+    int has_property(zend_object *zobj, zend_string *name, int has_set_exists, void **cache_slot)
+    void unset_property(zend_object *zobj, zend_string *name, void **cache_slot)
+    zval *get_property_ptr_ptr(zend_object *zobj, zend_string *name, int type, void **cache_slot)
+
+    These handlers correspond to the ``__get``, ``__set``, ``__isset`` and ``__unset`` methods. ``get_property_ptr_ptr``
+    is the internal equivalent of ``__get`` returning by reference. ``cache_slot`` is used to store the property
+    offset and ``zend_property_info``. ``zval *rv`` in ``read_property`` provides a place for temporary zvals that
+    are not stored in the object itself, like results of calls to ``__get``.
```

Comment:
I think adding something like this would be helpful here, because `rv` ownership is the main tricky part:

> `read_property` may directly return a zval owned by the object, in which case its reference count should not be modified by `read_property`, and the caller should not release it. Alternatively, it may return `rv`, in which case the refcount should be incremented, and the caller is responsible for releasing the value.